lib/cdo/throttle.rb

```diff
@@ -36,10 +46,25 @@ def self.throttle(id, limit, period, throttle_for = throttle_time)
     def self.empty_value
       {
         throttled_until: nil,
-        request_timestamps: []
+        request_entries: []
       }
     end
 
+    # Normalize old data format where we only stored request timestamps
+    # to the new format where we store request entries with timestamps and counts.
+    def self.normalize_value(raw_value)
+      if raw_value.key?(:request_entries)
+        raw_value
+      elsif raw_value.key?(:request_timestamps)
+        {
+          throttled_until: raw_value[:throttled_until],
+          request_entries: raw_value[:request_timestamps].map {|timestamp| {timestamp: timestamp, count: 1}}
+        }
+      else
+        raise 'Invalid throttle data format'
```

Comment:
Couldn't think of something more elegant to do here, but we really shouldn't be hitting this condition (famous last words).

Reply:
This seems ok to me, presumably we'd hit HoneyBadger here and have a way to quickly know if this is happening more than we expect?

Reply:
We could consider logging the error and just returning an empty value, in the interest of minimizing the chance of user impact. But yeah, I agree both that this seems unlikely and that we'll find out quickly if that expectation is wrong, so I'm not too worried one way or the other